config/config.go

```diff
@@ -395,7 +406,14 @@ func getDefaultStringFromFile(filename string, defaultValue string) (retVal stri
 
 // String marshals the given Config into a YAML string
```

Comment:
@jotak I haven't looked at the code that calls this String() function - the comment says it returns a YAML string... can you confirm that all the callers to that String() function only use the returned string for logging? i.e. the returned YAML is not used for additional processing of the YAML? I'm thinking maybe some unit tests do that??? I dunno.

Also, can you change the comment to this function -- in the comment, tell people this function is not to be used to process the YAML in any way since some of the data is obfuscated. We need people to know they cannot simply re-parse the returned YAML string and expect to get back to original YAML. The way around that would be to pass in a boolean argument to the String() function -- `obfuscate` -- and when true do what you are doing but if false don't call the new "Obfuscate" functions. I don't think we need this, though, unless there are callers to String() that need to re-parse the YAML for some reason.

Reply:
Finding actual callers of `String()` is not easy and changing the signature is not possible, for the same reason: it's actually an implementation of the `Stringer` interface and is used in a wide variety of stuff, like when you do something like `fmt.Sprintf("%v", obj)` the `String()` function of obj would be invoked. In our code we don't explicitly call `config.String()`. Much like java's `Object::toString()`.

But to be honest, this isn't really a problem, we *should* not be using such a formatting function for any business code. I mean, if we really wanted to read/parse as yaml, we would have something like `config.ToYAML()`, not `config.String()`, right? To me it is quite clear than a `String()` implementation is essentially used for logging purpose.

I tried to search in code for config occurrences that could be used in such a way as you describe, but didn't find any. I seriously doubt there's such a thing somewhere.

I think it was transformed into YAML for logging purpose because the native object logging in go was not satisfying for config (it doesn't show keys iirc)

PS: I'll update the function comment